File: tests/java/src/eu/ill/cameo/test/Result.java
/*
 * CAMEO
 *
 * Copyright 2015 Institut Laue-Langevin
 *
 * Licensed under BSD 3-Clause and GPL-v3 as described in license files.
 * You may not use this work except in compliance with the Licences.
 *
 */

package eu.ill.cameo.test;

import eu.ill.cameo.api.base.This;


public class Result {

	public static void main(String[] args) {

		This.init(args);
		
		This.setStringResult("this is a test result");
		This.terminate();
		
		System.out.println("Finished the application");
	}

}

File: tests/cpp/src/Result.cpp
/*
 * CAMEO
 *
 * Copyright 2015 Institut Laue-Langevin
 *
 * Licensed under BSD 3-Clause and GPL-v3 as described in license files.
 * You may not use this work except in compliance with the Licences.
 *
 */

#include <cameo/api/cameo.h>
#include <iostream>

using namespace std;
using namespace cameo;

int main(int argc, char *argv[]) {

	This::init(argc, argv);

	cout << "Waiting 1s..." << endl;
	this_thread::sleep_for(chrono::seconds(1));

	This::setResult("test result");

	cout << "Finished the application" << endl;

	This::terminate();

	return 0;
}
